src/tallies/filter_energy.cpp

```diff
@@ -0,0 +1,174 @@
+#include "openmc/tallies/filter_energy.h"
+
+#include "openmc/capi.h"
+#include "openmc/constants.h"  // For F90_NONE
+#include "openmc/mgxs_interface.h"
+#include "openmc/search.h"
+#include "openmc/settings.h"
+#include "openmc/xml_interface.h"
+
+namespace openmc {
+
+//==============================================================================
+// EnergyFilter implementation
+//==============================================================================
+
+void
+EnergyFilter::from_xml(pugi::xml_node node)
+{
+  auto bins = get_node_array<double>(node, "bins");
+  this->set_bins(bins);
+}
+
+void
+EnergyFilter::set_bins(gsl::span<const double> bins)
+{
+  // Clear existing bins
+  bins_.clear();
+  bins_.reserve(bins.size());
+
+  // Copy bins, ensuring they are valid
+  for (gsl::index i = 0; i < bins.size(); ++i) {
```

Comment:
Nit:  The established pattern in this commit is to use post increment instead of pre increment.